src/yang.act

```diff
@@ -2,61 +2,165 @@
 import yang.parser
 import yang.schema
 
+def eq_optional[T(Eq)](a: ?T, b: ?T) -> bool:
+    return a is not None and b is not None and a == b or a is None and b is None
+
+def optional_str[T](v: ?T, default: str = "None") -> str:
+    return str(v) if v is not None else default
+
+class NameRevMap[T](object):
```

Comment:
I think this is quite similar in purpose to the `Context` in yang.schema. One of the key features in `Context` is that we index modules with and without revision, so we can lookup the latest without any revision (common for imports), yet allow multiple revisions to exist.

I have been wavering a bit back and forth how to use such a context and in the current version I had straight up dicts for the source modules / submodules. Previously I did use it for the sources. I think I just hadn't wrapped my head around how a group of YANG modules need to be compiled. I have a much better idea now and I think it's worthwhile having both sources and the compiled output in such a structure, which is what we get now, though NameRevMap for sources and Context for compiled output. I think they could be the same.

Your use of extension is very elegant of course! Pro shit yo =)